network.go

```diff
@@ -1148,3 +1149,25 @@ func (n *network) IpamConfig() (string, []*IpamConf, []*IpamConf) {
 
 	return n.ipamType, v4L, v6L
 }
+
+func (n *network) IpamInfo() ([]*IpamInfo, []*IpamInfo) {
+	n.Lock()
```

Comment:
@aboch thanks for your comment, but I think the deep copy is needed here, I have already tested your idea, I ran `docker network inspect net01`, the cmd would hang all the time without output. If not deep copy, the memory stack will be deleted twice, so the program crashed.